### ISSUE TITLE
[BUG] missing `html` language methods should be implemented?

### ISSUE THREAD

**Issue by andreww2012** [yeonjuan/html-eslint#488]
**Describe the bug**

Hello! I am using this plugin together with [`eslint-plugin-html`](https://www.npmjs.com/package/eslint-plugin-html) to lint inline scripts.

Previously, how I configured the HTML parser like this:

```ts
import eslintParserHtml from '@html-eslint/parser';

{
  languageOptions: {
    parser: eslintParserHtml,
  },
  // (Of course, I'm adding the plugin too)
}
```

But since the plugin now provides an ESLint Language, the setup can be simplified by using the `language` property instead:

```ts
{
  language: '@html-eslint/html'
}
```

While there's another problematic aspect of this setup being the fact that [`eslint-plugin-html` doesn't work altogether with this setup](https://github.com/BenoitZugmeyer/eslint-plugin-html/issues/312) (which I managed to work around with a patch to `eslint-plugin-html` - but that's a different story), the actual issue I'm reporting here is that many "regular" ESLint rules crash on HTML files. Example of the error:

```
Oops! Something went wrong! :(

ESLint: 9.39.2

TypeError: Error while loading rule 'no-irregular-whitespace': sourceCode.getAllComments is not a function
Occurred while linting <...>/index.html
```

This happens because [`HTMLSourceCode` doesn't implement `getAllComments` method in particular](https://github.com/yeonjuan/html-eslint/blob/b7f3d1d94a5583a1540fdb6e8a1a982cd6445dca/packages/eslint-plugin/lib/languages/html-source-code.js#L36) and well as many other `SourceCode` methods in general.

I want to say the root cause is [a gap in ESLint language docs](https://eslint.org/docs/latest/extend/languages) which doesn't mention many of these methods at all but at the same time [state that `sourceCode` object certainly provides them](https://eslint.org/docs/latest/extend/custom-rules#accessing-the-source-code).

I'd like to know your option on would it make sense to add these missing methods to `HTMLSourceCode` class. The full `SourceCode` interface [can be found here](https://github.com/eslint/eslint/blob/11644b1dc2bdf4c4f3a97901932e5f25c9f60775/lib/types/index.d.ts#L248). Thank you.

**To Reproduce**

https://stackblitz.com/edit/vitejs-vite-yn6rwbap

Run `pnpm test` to run ESLint on `index.html` file. Expect it to crash because `language` is set.

Comment out `language` and uncomment `parser` property. Re-run the same command and it will not crash.

**Expected behavior**

Non-HTML rules do not crash when used on HTML files.

**Additional context**

**Comment by yeonjuan** [comment]
Hi @andreww2012 
The `no-irregular-whitespace` core rule is a rule designed for JavaScript.
Therefore, it cannot be used for `**/*.html` files.
As you can see from the [implementation of no-irregular-whitespace](https://github.com/eslint/eslint/blob/11644b1dc2bdf4c4f3a97901932e5f25c9f60775/lib/rules/no-irregular-whitespace.js#L270-L277), it is built to operate on the JavaScript AST.

I may have missed something in the SourceCode implementation. However, even if that were implemented, the generated AST is not a JavaScript AST, so it would be difficult to expect it to work correctly.

**Comment by andreww2012** [comment]
>  The `no-irregular-whitespace` core rule is a rule designed for JavaScript.

Sure; I'm enabling these rules solely to work on inline scripts - that's how `eslint-plugin-html` works. 

Maybe it would be possible to implement these methods as stubs? I'm happy to submit a PR implementing them if you approve this idea 🙂

**Comment by yeonjuan** [comment]
PRs are always welcome! However, even if it doesn't generate errors, it probably won't work with other rules. It might be good to look into how eslint-plugin-html works (probably this weekend). Of course, PRs to help me understand are welcome.
